src/mapping/grouping/cluster_config.jl

```diff
@@ -0,0 +1,202 @@
+calculating the distance of each point to its k-nearest-neighbors (with `k=min_neighbors`)
+and finding the distance corresponding to the highest derivative in the curve of the
+distances, sorted in ascending order. This distance is chosen as the optimal radius. It is
+described in [^Kriegel1996] and [^Schubert2017]. It typically performs considerably worse
+than the `"silhouette"` methods.
+
+[^Stender2021]:
+    Stender & Hoffmann 2021, [bSTAB: an open-source software for computing the basin
+    stability of multi-stable dynamical systems](https://doi.org/10.1007/s11071-021-06786-5)
+[^Gelbrecht2021]:
+    Maximilian Gelbrecht et al 2021, Monte Carlo basin bifurcation analysis,
+    [2020 New J. Phys.22 03303](http://dx.doi.org/10.1088/1367-2630/ab7a05)
+[^Kriegel1996]: Ester, Kriegel, Sander and Xu: A Density-Based Algorithm for Discovering
+    Clusters in Large Spatial Databases with Noise
+[^Schubert2017]:
+    Schubert, Sander, Ester, Kriegel and Xu: DBSCAN Revisited, Revisited: Why and How You
+    Should (Still) Use DBSCAN
+"""
+struct GroupViaClustering{R<:Union{Real, String}, M<:Metric, F<:Function} <: GroupingConfig
+    clust_distance_metric::M
+    min_neighbors::Int
+    rescale_features::Bool
+    optimal_radius_method::R
+    num_attempts_radius::Int
+    silhouette_statistic::F
+    max_used_features::Int
+end
+
+function GroupViaClustering(;
+        clust_distance_metric=Euclidean(), min_neighbors = 10,
+        rescale_features=true, num_attempts_radius=100,
+        optimal_radius_method::Union{Real, String} = "silhouettes_optim",
+        silhouette_statistic = mean, max_used_features = 0,
+    )
+    return GroupViaClustering(
+        clust_distance_metric, min_neighbors,
+        rescale_features, optimal_radius_method,
+        num_attempts_radius, silhouette_statistic, max_used_features
+    )
+end
+
+#####################################################################################
+# API funtion (group features)
+#####################################################################################
+function group_features(features::Vector{<:AbstractVector}, config::GroupViaClustering)
+    (; min_neighbors, clust_distance_metric, rescale_features, optimal_radius_method,
+    num_attempts_radius, silhouette_statistic, max_used_features) = config
+
+    nfeats = length(features); dimfeats = length(features[1])
+    if dimfeats ≥ nfeats
+        throw(ArgumentError("""
+        Not enough features. The algorithm needs the number of features
+        $nfeats to be greater or equal than the number of dimensions $dimfeats
+        """))
+    end
+
+    if rescale_features
+        features = rescale_to_01(features)
+    end
+
+    # Obtain optimal radius
+    if optimal_radius_method isa String
+        if max_used_features == 0 || max_used_features > nfeats
+            features_for_optimal = features
+        else
+            features_for_optimal = StatsBase.sample(features, max_used_features; replace = false)
```

Comment:
This line gave me trouble with some feature vectors. I used: 
```suggestion
          ind = sample(1:nfeats, minimum([length(features), max_used_features]); replace = false)
          features[:,ind]
```

Reply:
It's because for you `features` was a matrix. I left a comment on your PR that sample didn't work like that and you had to sample before you made them into matrix. Now, we no longer do matrices at all, so the way I sample is correct because it is done on a vector.